### ISSUE TITLE
Variable/argument type validation

### ISSUE THREAD

**Issue by swalkinshaw** [rmosolgo/graphql-ruby#816]
It looks like validation for variables and arguments aren't catching some instances of type mismatches.

Here's a link to an example in Launchpad (via graphql-js) where it fails: https://launchpad.graphql.com/x1l9nv7pl

> "Variable \"$sizes\" of type \"[ImageSize]\" used in position expecting type \"ImageSize!\".",

And here's the steps to reproduce in this gem:

Schema:
```graphql
schema {
  query: QueryRoot
}

input ImageSize {
  height: Int
  width: Int
  scale: Int
}

type QueryRoot {
  imageUrl(height: Int, width: Int, size: ImageSize, sizes: [ImageSize!]): String!
}
```

query:
```graphql
query ($sizes: [ImageSize]) {
  imageUrl(sizes: [$sizes])
}
```

```ruby
MySchema = GraphQL::Schema.from_definition(schema_def)
validator = GraphQL::StaticValidation::Validator.new(schema: MySchema)
query_string = "query ($sizes: [ImageSize]) {
  imageUrl(sizes: [$sizes])
}"
query = GraphQL::Query.new(MySchema, query_string, variables: { 'sizes' => [{ 'scale' => 2 }, { 'height' => 10, 'width' => 10, 'scale' => 1}] })
errors = validator.validate(query)[:errors] # []
query.valid? # true
query.validation_errors # []
query.variables.errors # []
```

**Comment by swalkinshaw** [comment]
I edited the original issue to be more clear 👌

**Comment by rmosolgo** [comment]
I wonder if this is throwing it off, where variables _without default values_ are treated as if they are non-null variables: 

https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/static_validation/rules/variable_usages_are_allowed.rb#L46-L48

**Comment by swalkinshaw** [comment]
It actually appears to be here: https://github.com/rmosolgo/graphql-ruby/blob/952bb07367a810dbedf930b5133a19aba9ffd2f3/lib/graphql/static_validation/rules/variable_usages_are_allowed.rb#L16

At this point `node.value` is an array, not `GraphQL::Language::Nodes::VariableIdentifier)`.

So any argument within a list/array is not getting validated? 😔

**Comment by rmosolgo** [comment]
:S :S :S

**Comment by rmosolgo** [comment]
Oh, this was fixed in 1.6.5, thanks again for debugging this!
